cdap-ui/app/cdap/components/PipelineList/DeployedPipelineView/RunsCount/index.tsx

```diff
@@ -15,27 +15,28 @@
  */
 
 import * as React from 'react';
-import { connect } from 'react-redux';
-import { IRunsCountMap, IPipeline } from 'components/PipelineList/DeployedPipelineView/types';
+import { IApplicationRecord } from 'components/PipelineList/DeployedPipelineView/types';
+import {
+  getProgram,
+  getProgramRuns,
+} from 'components/PipelineList/DeployedPipelineView/graphqlHelper';
 
 interface IProps {
-  runsCountMap: IRunsCountMap;
-  pipeline: IPipeline;
+  pipeline: IApplicationRecord;
 }
 
-const RunsCountView: React.SFC<IProps> = ({ runsCountMap, pipeline }) => {
-  const runsCount = runsCountMap[pipeline.name] || 0;
+const RunsCountView: React.SFC<IProps> = ({ pipeline }) => {
+  const runsCount = getRunsCount(pipeline);
 
   return <div className="runs">{runsCount}</div>;
 };
 
-const mapStateToProps = (state, ownProp) => {
-  return {
-    runsCountMap: state.deployed.runsCountMap,
-    pipeline: ownProp.pipeline,
-  };
-};
+function getRunsCount(pipeline) {
+  const program = getProgram(pipeline);
+
+  return getProgramRuns(program).length;
```

Comment:
didn't you add the `totalRuns` field in graphQL? If so, this RunsCount should be using that

Reply:
I did not add that field to the ProgramRecord type since it is inefficient to make a separate query to the backend to fetch information that we already have (i.e., the length of the list of run records). 

To get the field `runs`, we use this endpoint `/v3/namespaces/${namespace}/runs`, which does not have limits on the number of entries that it returns. 

Is there a reason to make a separate query to get the `totalRuns`?

Reply:
you have to use that totalRuns. The runs endpoint is limited to 100 runs.

Reply:
I added the field, implemented the resolver, updated the query, and displayed the results in the UI